src/main/java/com/google/sps/data/Keywords.java

```diff
@@ -0,0 +1,129 @@
+package com.google.sps.data;
+
+import java.io.IOException;
+import com.google.cloud.language.v1.Document;
+import com.google.cloud.language.v1.LanguageServiceClient;
+import com.google.cloud.language.v1.AnalyzeEntitiesRequest;
+import com.google.cloud.language.v1.AnalyzeEntitiesResponse;
+import com.google.cloud.language.v1.EncodingType;
+//import com.google.cloud.language.v1.Entity;
+import com.google.cloud.vision.v1.EntityAnnotation;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.TreeSet;
+
+/**
+ * Provide methods for the storage, extraction, and retrieval of keywords.
+ */
+public final class Keywords {
+
+  private static final int MAX_NUM_KEYWORDS = 10;
+
+  /**
+   * @return a list of the 10 most salient keywords
+   */
+  public static List<String> getKeywords(String key) throws EntityNotFoundException {
+    Key datastoreKey = KeyFactory.stringToKey(key);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Entity result = datastore.get(datastoreKey);
+    
+    Collection<String> keywordCollection = (Collection<String>) result.getProperty("keywords");
+    List<String> keywordList = new ArrayList<>();
+    int index = 0;
+    for (String keyword : keywordCollection) {
+      keywordList.add(keyword);
+    
+      index++;
+      if (index >= MAX_NUM_KEYWORDS) {
+        break;
+      }
+    }
```

Comment:
[Nit] Use a filter in the query instead of doing it manually.

Reply:
I'm not using a query anymore. Instead, I'm using the entity key to directly retrieve the entity from the datastore, rather than querying it. The entity list contains the full keyword list (rather than truncating it down to 10 keywords), so I think that this is still required.

Reply:
Makes sense.